rdbms/src/main/java/org/spine3/server/storage/jdbc/event/JdbcEventStorage.java

```diff
@@ -137,13 +165,142 @@ public void close() throws DatabaseException {
         try {
             super.close();
         } catch (Exception e) {
-            throw propagate(e);
+            throw wrapped(e);
         }
         closeAll(iterators);
         iterators.clear();
         dataSource.close();
     }
 
+    /**
+     * Predicate matching an {@link EventStorageRecord} to a number of {@link EventFilter} instances.
+     */
+    private static class EventRecordPredicate implements Predicate<EventStorageRecord> {
```

Comment:
Now we have a predicate similar to this in `core-java` — `EventStorage.MatchesStreamQuery`, so, maybe, we should split that predicate and reuse the part we need (as it filters through more broad set of attributes).